### PR TITLE
Internal: add Cypress integration tests

### PR DESCRIPTION
**Pull request by christianvuerings**
Instead of Netlify CLI, use GitHub actions to run cypress integration tests.

## Test Plan

* GitHub Integration Tests action pass

## Next steps
- [x] Add https://www.npmjs.com/package/cypress-axe to enable accessibility integration tests 
- [x] Record tests on Cypress dashboard

## Notes
* Cypress does not have flow-typed definitions so we are ignoring the directory for now.
* We use the official Cypress GitHub Action: https://github.com/cypress-io/github-action.
* Tests will only be recorded when we push to master (otherwise we might leak the Cypress key)

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "baseUrl": "http://localhost:3000",
+  "projectId": "x99ctf",
```
Comment (christianvuerings) [inline cypress.json:3]: See https://dashboard.cypress.io/projects/x99ctf - plan to add this link to our docs once this is merged

File: docs/src/Installation.doc.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ card(
       <Link inline target="blank" href="https://npmjs.org/package/gestalt">
         <img
           src="https://img.shields.io/npm/v/gestalt.svg?label=gestalt"
-          alt="Gestalt's NPM package version badge"
+          alt="Gestalt NPM package version badge"
```
Comment (christianvuerings) [inline docs/src/Installation.doc.js:19]: A11Y fix - axe-core picked up that we had the same descriptions for these links

File: docs/src/components/App.js [hunk 3]
```diff
@@ -62,12 +62,18 @@ export default function App(props: Props): Node {
             </Box>
             <Divider />
             <Box width="100%">
-              <Box padding={4} mdPadding={6} lgPadding={8} width="100%">
+              <Box
+                padding={4}
+                mdPadding={6}
+                lgPadding={8}
+                width="100%"
+                role="main"
```
Comment (christianvuerings) [inline docs/src/components/App.js:70]: A11Y fix - we need to add `role` to all places that have content on the page

File: docs/src/components/Card.js [hunk 4]
```diff
@@ -38,8 +38,12 @@ export default function Card({
             data-anchor
           >
             <Row alignItems="baseline" gap={1}>
-              {name}
-              <Link href={`#${slugifiedId}`} inline>
+              <Box>{name}</Box>
+              <Link
+                href={`#${slugifiedId}`}
+                inline
+                accessibilityLabel={`${name} - Anchor tag`}
```
Comment (christianvuerings) [inline docs/src/components/Card.js:45]: A11Y fix: used to be an empty link without a good label

File: docs/src/components/Markdown.css [hunk 5]
```diff
@@ -1,5 +1,5 @@
 .Markdown pre {
-  background-color: rgb(236, 236, 236);
+  background-color: rgba(236, 236, 236, 0.5);
```
Comment (christianvuerings) [inline docs/src/components/Markdown.css:2]: A11Y fix - detected that the contrast wasn't high enough (WCAG 2.0 AA)

File: docs/src/components/SidebarSection.js [hunk 6]
```diff
@@ -10,8 +10,8 @@ export default function SidebarSection({
   section: sidebarIndexType,
 |}): Node {
   return (
-    <>
-      <Box padding={2} marginTop={4} role="list">
+    <Box role="list">
```
Comment (christianvuerings) [inline docs/src/components/SidebarSection.js:13]: A11Y Fix: we previous had a `<Box role="list" />` without any `role="listitem"` in them.

File: package.json [hunk 7]
```diff
@@ -50,20 +53,20 @@
     "lint-staged": "^10.3.0",
     "netlify-cli": "^2.61.2",
     "papaparse": "^5.3.0",
+    "postcss": "^7.0.32",
```
Comment (christianvuerings) [inline package.json:56]: alphabetized these (happened when I added cypress as a dev package)

**Review by AlbertCarreras — APPROVED** [review]

**Pull request merged**